### PR TITLE
[DOCS]Documenting id in selection docs.

### LINKED ISSUE #3359: Document id selection keyword
## Is your feature request related to a problem? ##

The selection docs have `bynum` but not `id`. `id` is more helpful in the sense that it actually selects the right `id` values, unlike `bynum` which is just `index` + 1.

## Describe the solution you'd like ##

Document `id`.

### PR DESCRIPTION
**Pull request by umak1106**
Fixes #3359 

Changes made in this Pull Request:
 - Documenting id selection keyword

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Review by ojeda-e — CHANGES_REQUESTED** [review]
Hi @umak1106.
Thanks for submitting this PR with the description of the changes submitted as well as referencing the issue! Reviewers appreciate it!

Keep in mind that here we are writing documentation for users of MDAnalysis. When writing docs, the aim is to communicate the functionality of the code provided instead of giving the definition/description of a given element. In other words, what we want to tell the reader is how to use our code. In this case, we want to tell the reader how to use the selection commands with the `id` topology attribute. In this sense, I would start with the functionality of the selection id, i.e.:
`selects all atoms in the given range which consists of two numbers seperated by a colon such as ``id 5:10`` (both inclusive).` 
 
For consistency, it would be nice if the description of `id` matches the one we already have for `index` and `bynum`. Specifically, it would be nice to add a sentence like "All atoms in the MDAnalysis.Universe are consecutively numbered, and the _id_ runs from ...". 

As suggested by @orbeckst [here](https://github.com/MDAnalysis/mdanalysis/issues/3359#issuecomment-888709427), it would be helpful to add a brief comment on the difference between _id_ and _index_. 
Maybe consider adding a note at the end of 3.1.6 to show the difference between id and index? (I am not entirely sure that's the best way to document the difference, just giving you ideas).

I also highlighted some typos below.

File: package/doc/sphinx/source/documentation_pages/selections.rst [hunk 1]
```diff
@@ -300,7 +300,15 @@ bynum *index-range*
     selects atoms 5 through 10 inclusive. All atoms in the
     :class:`MDAnalysis.Universe` are consecutively numbered, and the index
     runs from 1 up to the total number of atoms.
-
+    
+id *index-range*
+    id is a topology attribute which can be labeled according to the user
+    and conventionally corresponds to serial number in PDB format.
+    selects all atoms in the given range which consists of two numbers 
+    seperated by a colon such as ``id 5:10`` (both inclusive).
```
Comment (ojeda-e) [inline package/doc/sphinx/source/documentation_pages/selections.rst]:
```suggestion
    separated by a colon such as ``id 5:10`` (both inclusive).
```

File: package/doc/sphinx/source/documentation_pages/selections.rst [hunk 2]
```diff
@@ -300,7 +300,15 @@ bynum *index-range*
     selects atoms 5 through 10 inclusive. All atoms in the
     :class:`MDAnalysis.Universe` are consecutively numbered, and the index
     runs from 1 up to the total number of atoms.
-
+    
+id *index-range*
+    id is a topology attribute which can be labeled according to the user
+    and conventionally corresponds to serial number in PDB format.
+    selects all atoms in the given range which consists of two numbers
```
Comment (ojeda-e) [inline package/doc/sphinx/source/documentation_pages/selections.rst]: Capitalize the first word of a document and the first word after a period.

```suggestion
    Selects all atoms in the given range which consists of two numbers 
```

Comment (umak1106) [reply package/doc/sphinx/source/documentation_pages/selections.rst]: Actually it wasn't capitalized in bynum and index so i left it the same

File: package/doc/sphinx/source/documentation_pages/selections.rst (hunk 1 above)
Comment (umak1106) [reply package/doc/sphinx/source/documentation_pages/selections.rst]: Done

**Review by ojeda-e — CHANGES_REQUESTED** [review]
Thanks for working on the suggestions and requesting the review, @umak1106.

I have some additional suggestions, mostly to be consistent with the docs for _index_ and _bynum_.

- In the description of index, what do you think about something like:
`selects all atoms within a range of (1-based) inclusive indices. The range consists of two numbers separated by a colon, e.g. (your example)`. 

- What do you think about this for the second part: 
`All atoms in the MDAnalysis.Universe are consecutively numbered, ~~though
    id (an topology attribute)~~ and the index runs from ...` 

It would keep the style from `bynum` and `index`, hence consistent. 

Another important point to consider is that docs are usually succinct. The information provided should be concise: short but effective. (Sorry I omitted this comment in the previous review. This is actually a very relevant point and may help you to write docs in the future.)

I acknowledge your intention of highlighting that id can be assigned by the user. IMO, that would likely read better in the note. What do you think about moving that part? Not asking you to move it, this is also my personal opinion. This is more for you to think about it and consider what would be best for the user. If you come to a good justification, please share it with us.

Minor comment:
- There is a comma before the "e.g.", not a period.

If you would like to discuss suggestions, feel free to post it here to participate in the conversation. Thanks.

File: package/doc/sphinx/source/documentation_pages/selections.rst [hunk 3]
```diff
@@ -300,13 +300,31 @@ bynum *index-range*
...
 index *index-range*
     selects all atoms within a range of (0-based) inclusive indices,
     e.g. ``index 0`` selects the first atom in the universe; ``index 5:10``
     selects atoms 6 through 11 inclusive. All atoms in the
     :class:`MDAnalysis.Universe` are consecutively numbered, and the index
     runs from 0 up to the total number of atoms - 1.
+    
+    
+.. note::
+    While ``index`` is the underlying position of an atom in a linear array
```
Comment (ojeda-e) [inline package/doc/sphinx/source/documentation_pages/selections.rst]: I am not convinced that the line "is the underlying position of an atom in a linear array of atoms in the universe" is a good fit here. The description is a bit abstract and the user often prefers practical information. I think the note should briefly say the difference between id and index in a way that is practical from the user's perspective.

File: package/doc/sphinx/source/documentation_pages/selections.rst [hunk 4]
```diff
@@ -300,13 +300,31 @@ bynum *index-range*
...
     runs from 1 up to the total number of atoms.
-
+    
+id *index-range*
+    selects all atoms in the given range which consists of two numbers 
+    separated by a colon such as ``id 5:10`` (both inclusive).
+    e.g. ``id 1`` selects the atom with id 1; ``id 5:10`` selects the
+    atoms with ids 5 through 10 both inclusive. All atoms in the
+    :class:`MDAnalysis.Universe` are consecutively numbered, though
+    id (an topology attribute) can be labeled by the user and need
```
Comment (lilyminium) [inline package/doc/sphinx/source/documentation_pages/selections.rst]: Thanks for addressing this, @umak1106 and @ojeda-e. I just wanted to chime in here and say that while it's good to distinguish between `id` and `index`, IMO the more important distinction is between `id` and `bynum`. `id` commonly starts at 1, so it looks a lot like `bynum` -- however, they are fundamentally different. `bynum` is equivalent to `index + 1`, whereas `id` is truly a label; there is nothing stopping it from going `[10, 1, 4, 8, 1]`. Because of this, I would be keen to avoid confusion with this note: 

> All atoms in the
> :class:`MDAnalysis.Universe` are consecutively numbered"

IMO the section below does not need to be repeated either. The range selection acts like this for all ranges. They are spelled out for the `index`, `bynum`, `resid`, and `resnum` because each keyword has slightly different and unexpected behaviour. It is again potentially confusing by implying that only one atom can have `id=1`, where in fact multiple atoms can have the same `id`.

> selects all atoms in the given range which consists of two numbers 
> separated by a colon such as ``id 5:10`` (both inclusive).
> e.g. ``id 1`` selects the atom with id 1; ``id 5:10`` selects the

Comment (ojeda-e) [reply package/doc/sphinx/source/documentation_pages/selections.rst]: Thanks for the input @lilyminium I was about to pin you for a review here.

**Comment by lilyminium** [comment]
@umak1106 I think this is the third PR you have opened to MDAnalysis. All your contributions are great and we would welcome more, but can I please ask you to stop addressing issues tagged with `outreachy-task` or `gsoc-starter`? These are issues that we have identified as good starter issues for outreachy or GSOC applicants, and there is a limited number of them. It would be great if you could leave them for other applicants and look at different issues instead.

**Comment by umak1106** [comment]
> @umak1106 I think this is the third PR you have opened to MDAnalysis. All your contributions are great and we would welcome more, but can I please ask you to stop addressing issues tagged with `outreachy-task` or `gsoc-starter`? These are issues that we have identified as good starter issues for outreachy or GSOC applicants, and there is a limited number of them. It would be great if you could leave them for other applicants and look at different issues instead.

Ok mentor , I will take a look at the other issues .

File: package/doc/sphinx/source/documentation_pages/selections.rst (hunk 4 above)
Comment (umak1106) [reply package/doc/sphinx/source/documentation_pages/selections.rst]: I updated it according to your review . Could you please take a look at it again

Comment (umak1106) [reply package/doc/sphinx/source/documentation_pages/selections.rst]: ?

File: package/doc/sphinx/source/documentation_pages/selections.rst [hunk 5]
```diff
@@ -300,14 +300,29 @@ bynum *index-range*
...
     e.g. ``index 0`` selects the first atom in the universe; ``index 5:10``
     selects atoms 6 through 11 inclusive. All atoms in the
     :class:`MDAnalysis.Universe` are consecutively numbered, and the index
     runs from 0 up to the total number of atoms - 1.
-
+    
+    
+.. note::
+    Conventionally ``id`` corresponds to the serial number in PDB format and so
+    starts from 1 like the bynum. But ``id`` can be labeled by the user and need
```
Comment (ojeda-e) [inline package/doc/sphinx/source/documentation_pages/selections.rst]:
```suggestion
    starts from 1 like the ``bynum``. But ``id`` can be labeled by the user and need 
```

**Review by ojeda-e — COMMENTED** [review]
Hi @umak1106, 
is looking good so far! I think the text in the description is very clear. Thanks for considering the suggestions.

Quick ping to @lilyminium who may give you additional suggestions.

**Review by ojeda-e — CHANGES_REQUESTED** [review]
Hi @umak1106,
I think the note can be further improved to deliver a clearer message, it still reads a bit raw. I included some suggestions and questions below.

File: package/doc/sphinx/source/documentation_pages/selections.rst [hunk 6]
```diff
@@ -300,14 +300,29 @@ bynum *index-range*
...
 index *index-range*
     selects all atoms within a range of (0-based) inclusive indices,
     e.g. ``index 0`` selects the first atom in the universe; ``index 5:10``
     selects atoms 6 through 11 inclusive. All atoms in the
     :class:`MDAnalysis.Universe` are consecutively numbered, and the index
     runs from 0 up to the total number of atoms - 1.
-
+    
+    
+.. note::
```
Comment (ojeda-e) [inline package/doc/sphinx/source/documentation_pages/selections.rst:317]: I think I would rephrase the note similar to this:
Conventionally, ``id`` corresponds to the serial number in PDB format. In contrast to ``bynum``, the ``id`` index is not necessarily continuous nor unique, and can be arbitrarily assigned by the user. Therefore, different atoms in the :class:`MDAnalysis.Universe` can share the same ``id`` and the user can set the ``id`` index to an arbitrary number.
What do you think? Do you think there is somethin missing here?

File: package/doc/sphinx/source/documentation_pages/selections.rst [hunk 7]
```diff
@@ -300,14 +300,29 @@ bynum *index-range*
     selects atoms 5 through 10 inclusive. All atoms in the
     :class:`MDAnalysis.Universe` are consecutively numbered, and the index
     runs from 1 up to the total number of atoms.
-
+    
+id *index-range*
+    selects all atoms in the given range which consists of two numbers 
+    separated by a colon such as ``id 5:10`` (both inclusive),
+    e.g. ``id 1`` selects all the atoms with id 1; ``id 5:7`` selects all
+    atoms with ids 5, all atoms with ids 6 and all atoms with ids 7.
```
Comment (ojeda-e) [inline package/doc/sphinx/source/documentation_pages/selections.rst]: Out of curiosity, why did you omit to mention that by default the `id` selection is 1-based? The id as 0-based makes sense only after it is set by the user. Otherwise, the selection will be empty. Example:
```Python3
>>> import MDAnalysis as mda
>>> from MDAnalysis.tests.datafiles import PDB
>>> u = mda.Universe(PDB)
>>> print(u.select_atoms('id 0'))
<AtomGroup []>
>>> print(u.select_atoms('id 1'))
<AtomGroup [<Atom 1: N of type N of resname MET, resid 1 and segid SYSTEM and altLoc >]>
>>> u.atoms.ids = 0
>>> print(u.select_atoms('id 0'))
<AtomGroup [<Atom 1: N of type N of resname MET, resid 1 and segid SYSTEM and altLoc >, ...
```

Comment (umak1106) [reply package/doc/sphinx/source/documentation_pages/selections.rst]: No actually i did mention  in the note that it starts from 1 so i didn't add it here but i made the changes now.

**Review by ojeda-e — APPROVED** [review]
Thanks, @umak1106!
It looks good to me. I am approving this PR since the description of the `id` selection and a note with a brief explanation of the difference between `id` and `bynum` were added.

I will ping @lilyminium to have a look in case I am missing something (and to check if an example would be a good fit there).

File: package/doc/sphinx/source/documentation_pages/selections.rst [hunk 8]
```diff
@@ -300,14 +300,26 @@ bynum *index-range*
...
     selects all atoms within a range of (0-based) inclusive indices,
     e.g. ``index 0`` selects the first atom in the universe; ``index 5:10``
     selects atoms 6 through 11 inclusive. All atoms in the
     :class:`MDAnalysis.Universe` are consecutively numbered, and the index
     runs from 0 up to the total number of atoms - 1.
-
+    
+    
+.. note::
+    Conventionally ``id`` corresponds to the serial number in PDB format. In contrast to ``bynum``, the ``id`` index is not
```
Comment (lilyminium) [inline package/doc/sphinx/source/documentation_pages/selections.rst]:
```suggestion
    Conventionally, ``id`` corresponds to the serial number in the PDB format. In contrast to ``bynum``, the ``id`` topology attribute is not 
```

File: package/doc/sphinx/source/documentation_pages/selections.rst [hunk 9]
```diff
@@ -300,14 +300,26 @@ bynum *index-range*
...
     e.g. ``index 0`` selects the first atom in the universe; ``index 5:10``
     selects atoms 6 through 11 inclusive. All atoms in the
     :class:`MDAnalysis.Universe` are consecutively numbered, and the index
     runs from 0 up to the total number of atoms - 1.
-
+    
+    
+.. note::
+    Conventionally ``id`` corresponds to the serial number in PDB format. In contrast to ``bynum``, the ``id`` index is not 
+    necessarily continuous nor unique, and can be arbitrarily assigned by the user. Therefore, different atoms in the
```
Comment (lilyminium) [inline package/doc/sphinx/source/documentation_pages/selections.rst]:
```suggestion
    necessarily continuous, ordered, or unique, and can be arbitrarily assigned by the user.
```

Comment (lilyminium) [reply package/doc/sphinx/source/documentation_pages/selections.rst]: IMO you've already said the last sentence in the previous sentence.

**Review by lilyminium — COMMENTED** [review]
Thanks for the updates @umak1106 and @ojeda-e -- just some nitpicks with the wording of the note. I'll make it a non-blocking review so @ojeda-e can merge whenever she's happy :)

**Comment by umak1106** [comment]
@ojeda-e  
I have no idea why one of the checks is failing . Could you please look into it ?

**Comment by ojeda-e** [comment]
> @ojeda-e I have no idea why one of the checks is failing . Could you please look into it ?

Sure, I restarted the jobs. Once they pass and I'm happy to approve and merge. 

Thanks @umak1106 for this contribution!
Edit: Thanks to @lilyminium for the comments above too :)

**Review by ojeda-e — APPROVED** [review]
Looks good to me. I'm approving this PR.
